### ISSUE TITLE
Outgoing post requests  send(Request::post(...)) do not work

### ISSUE THREAD

**Issue by trobanga** [fermyon/spin#2080]
spin 2.0.0 (e4bb235 2023-11-03)
cloud 0.5.1 [installed]
js2wasm 0.6.1 [installed]
py2wasm 0.3.2 [installed]

I am trying to send an outgoing post request with some payload but the http::send() does not finish.

```rust
#[http_component]
async fn handle_notion_ical_fermyon(req: Request) -> Result<impl IntoResponse> {

    let body = "{data: [1]}";
    let request = RequestBuilder::new(Method::Post, "")

        .uri(&format!("http://localhost:3001",))
        .method(Method::Post)
        .body(body)
        .build();

    let response: IncomingResponse = send(request).await?;

    // unreachable

    let status = response.status();
    tracing::info!(?status);
 
    Ok(Response::builder()
        .status(200)
        .header("content-type", "text/plain")
        .body("Hello, Fermyon")?)
}
```

Inside send() it's stuck here
```rust
        let response = executor::outgoing_request_send(request)
            .await
            .map_err(SendError::Http)?;
```

**Comment by trobanga** [comment]
I wrote a dummy server that outputs incoming requests:
INFO dummy_server: req: Request { method: POST, uri: http://localhost:3001/, version: HTTP/1.1, headers: {"host": "localhost:3001", "transfer-encoding": "chunked"}, body: Body(Streaming) }
    at src/main.rs:9

 INFO dummy_server: Body(Streaming)
    at src/main.rs:12

And it seems the body is never streamed.

**Comment by lann** [comment]
Hello, thanks for the report. I have confirmed this problem and am looking into it.

**Comment by lann** [comment]
Also, please note that streaming support is experimental. We have been expecting hiccups. :slightly_smiling_face:

**Comment by dicej** [comment]
I think the issue here is that the `http_component` macro is a bit too flexible and is allowing function signatures which we didn't necessarily mean to support.  The intention was that `async` handlers would use a `wasi-http` style handler, e.g. `async fn handle_request(request: IncomingRequest, response_out: ResponseOutparam)`, which is well-tested.  Combining the non-streaming `Request` and `Response` types with `async` is not something we thought to try or support.

I'll dig into this to understand exactly where it's breaking, but the solution may be to simply disallow that combination in the macro.  Meanwhile, the workaround is to use the `wasi-http` types instead -- see the `wasi-http-rust-streaming-outgoing-body` example in this repository for reference.

**Comment by dicej** [comment]
Ok, nevermind what I said above -- looks like this scenario _is_ intended to work after all and I accidentally broke it.  Sorry about that!  I'll have a PR up to fix it shortly.

**Comment by lann** [comment]
The fix for this just went out in the 2.0.1 release.
